### PR TITLE
fix(float): remove -1 in height clamp

### PR DESCRIPTION
**Pull request by Diablo-D3**
Problem: Clamp for height in floating windows enforced no more than editor height - 1, disallowing full editor height floating windows when using cmdheight=0

Solution: Clamp to full height, removing the -1. Tested to give the intended results with cmdheight=0, 1, or more than 1. This also inadvertently fixes a rendering error with cmdheight >1 where the bottom border would be overlapped by the cmdline.

### CODE REVIEW

File: src/nvim/window.c [hunk 1]
```diff
@@ -59,6 +59,7 @@
 #include "nvim/move.h"
 #include "nvim/normal.h"
 #include "nvim/option.h"
+#include "nvim/option_defs.h"
```
Comment (clason) [inline src/nvim/window.c:62]: Why did you add this header? I don't see why your change should need that?

Comment (Diablo-D3) [reply src/nvim/window.c:62]: Ahh, that should not be in there. Whoops. This is a leftover when it was still "clamp to -cmdheight, not -1", and later became "just clamp"

Comment (clason) [reply src/nvim/window.c:62]: (I suspect it's left over from when you tried to use `cmdheight` instead of a static `1`.)

**Comment by clason** [comment]
Two tests in https://github.com/neovim/neovim/blob/master/test/functional/ui/float_spec.lua need to be adapted to the change in behavior.

**Review by zeertzjq — DISMISSED** [review]
As can be seen in the test failures, with 'cmdheight' = 1 the last of the float is invisible if it's at the bottom of the screen. Actually this problem already exited for 'cmdheight' > 1.

**Comment by Diablo-D3** [comment]
image
Re: the inadvertent fix of the missing border, my branch on the left, 0.8.3 on the right.

File: test/functional/ui/float_spec.lua [hunk 2]
```diff
@@ -3631,9 +3631,9 @@ describe('float window', function()
         ]], float_pos=expected_pos}
       else
         screen:expect([[
-                    {1:very           }               |
-          {0:~         }{1:^float          }{0:               }|
-                                                  |
+                    {1:such           }               |
+          {0:~         }{1:very           }{0:               }|
+                    ^                              |
```
Comment (zeertzjq) [inline test/functional/ui/float_spec.lua:3636]: This change defeats the purpose of this test. I think it's better to change this test to run with cmdheight = 0.
